source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -425,25 +425,22 @@ void SetupContextForAmazonServiceAccount(string @namespace, string clusterUrl, s
                 var clusterName = variables.Get(SpecialVariables.EksClusterName);
                 log.Info($"Creating kubectl context to {clusterUrl} (namespace {@namespace}) using EKS cluster name {clusterName}");
 
-                if (!TrySetAws())
-                {
-                    log.Warn("Could not find the aws cli, falling back to the aws-iam-authenticator.");
-                }
-                else
+                if (TrySetAws())
                 {
                     try
                     {
-                        var region = Regex.Match(clusterUrl, @"(us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d", RegexOptions.IgnoreCase).Value;
+                        var region = clusterUrl.Replace(".eks.amazonaws.com", "").Split('.').Last();
```

Comment:
1. Look at extract to methods
2. Look at extracting to encapsulated classes/logic